protocols/direct-funding_test.go

```diff
@@ -1,14 +1,93 @@
 package protocols
 
 import (
+	"math/big"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/types"
 )
 
+// TestNew tests the constructor using a TestState fixture
 func TestNew(t *testing.T) {
 	_, err := NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
 	if err != nil {
 		t.Error(err)
 	}
 }
+
+// Construct various variables for use in TestUpdate
+var s, _ = NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
+var dummySignature = state.Signature{
+	R: common.Hex2Bytes(`49d8e91bd182fb4d489bb2d76a6735d494d5bea24e4b51dd95c9d219293312d9`),
+	S: common.Hex2Bytes(`22274a3cec23c31e0c073b3c071cf6e0c21260b0d292a10e6a04257a2d8e87fa`),
+	V: byte(1),
+}
+var dummyStateHash = common.Hash{}
+var stateToSign state.State = s.ExpectedStates[0]
+var stateHash, _ = stateToSign.Hash()
+var privateKeyOfParticipant0 = common.Hex2Bytes(`caab404f975b4620747174a75f08d98b4e5a7053b691b41bcfc0d839d48b7634`)
+var correctSignatureByParticipant, _ = stateToSign.Sign(privateKeyOfParticipant0)
+
+func TestUpdate(t *testing.T) {
+	// First, prepare a new objective using the constructor:
+	s, _ := NewDirectFundingObjectiveState(state.TestState, state.TestState.Participants[0])
+
+	// Next, prepare an event with a mismatched channelId
+	e := ObjectiveEvent{
+		ChannelId: types.Destination{},
+	}
+	// Assert that Updating the objective with such an event returns an error
+	// TODO is this the behaviour we want? Below with the signatures, we prefer a log + NOOP (no error)
+	_, err := s.Update(e)
+	if err == nil {
+		t.Error(`ChannelId mismatch -- expected an error but did not get one`)
+	}
```

Comment:
```suggestion
	if _, err := s.Update(e); err == nil {
		t.Error(`ChannelId mismatch -- expected an error but did not get one`)
	}
```

This is a recommended style in order to minimize scope of variables. ie, prevent accidentally referring to `err` later in the same function when its meaning has become stale.